client/lib/utils/pdf-support.js

```diff
@@ -0,0 +1,36 @@
+import _ from 'lodash';
+
+/**
+ * This function result is cached for performance reasons, since browser capabilities won't change.
+ *
+ * @returns {string|false} "native" if the browser can display PDFs and code can reliably call JS
+ * functions on those PDFs (such as print()). "addon" if the browser can display PDFs, but there's
+ * no guarantee that they will respond to a DOM-like API. false if the browser can't display
+ * PDFs at all.
+ */
+export default _.memoize( () => {
+	if ( navigator.userAgent.includes( 'Firefox' ) ) {
```

Comment:
Is there any alternative to using `navigator.userAgent`?

MDN lists it as a deprecated feature that will be removed eventually. See: https://developer.mozilla.org/en-US/docs/Web/API/NavigatorID/userAgent